packages/playwright-core/src/server/registry/browserFetcher.ts

```diff
@@ -49,8 +50,10 @@ export async function downloadBrowserWithProgressBar(title: string, browserDirec
         debugLogger.log('install', `SUCCESS installing ${title}`);
         break;
       }
-      if (await existsAsync(zipPath))
+      if (await existsAsync(zipPath)) {
         await fs.promises.unlink(zipPath);
+        await fs.promises.rmdir(downloadFolder, { recursive: true });
```

Comment:
I think we should move this next to the line 70, to  nicely cleanup even on failure.

Reply:
Okay